src/xrCore/FS.cpp

```diff
@@ -137,7 +137,7 @@ bool file_handle_internal(pcstr file_name, u32& size, int& file_handle)
 
 void* FileDownload(LPCSTR file_name, const int& file_handle, u32& file_size)
 {
-    void* buffer = xr_malloc(file_size);
+    void* buffer = xr_malloc(file_size + 1); // IDK why need +1 for file size, but without this caused invalid write size 1 for this buffer
```

Comment:
It's strange. I guess this behaviour wasn't caused by your previous commits in this PR?  Looks like we should add ifdef for debug mode and force the compiler to print a message.

Reply:
This manifested itself long before this PR. Honestly, this place looks weird. Probably, the problem is in the wrong implementation of getting the file size, although I still could not detect differences in file size and buffer size.